doc/partitioned-db.md

```diff
@@ -0,0 +1,200 @@
+# Partitioned Database
```

Comment:
I feel like the partitioned database we talk about here and support is Cloudant only. Maybe it's better to have it in the title? or add some introductions like:
LooopBack 4 supports a partitioned database Cloudant....

Same for README file.

Reply:
@agnes512 The doc here is inside loopback-connector-cloudant only :) which is already the best note of the database it works with